ospf6d/ospf6_route.c

```diff
@@ -704,27 +704,6 @@ struct ospf6_route *ospf6_route_add(struct ospf6_route *route,
 	}
 
 	if (old) {
-		/* if route does not actually change, return unchanged */
-		if (ospf6_route_is_identical(old, route)) {
```

Comment:
Rather than entirely remove this, shouldn't it now become a simpler pointer comparison? If the comparison is always false *for different struct instances* as you say then we should still check if the instance is the same right?

Reply:
Yeah, I thought about this, but it's not possible either. `route` is always a new object (lock is asserted to be 0), and `old` is always an existing object (lock is decremented).